ui/src/services/core/device-control-service/device-control-service.ts

```diff
@@ -1,15 +1,16 @@
 import { socket } from '@/api/socket'
-import { TransactionService } from '@/services/core/transaction-service/transaction-service'
 
 import { KEYBOARD_KEYS_MAP } from '@/constants/keyboard-keys-map'
 
+import type { TransactionService } from '@/services/core/transaction-service/transaction-service'
+import type { DeviceBySerialStore } from '@/store/device-by-serial-store'
 import type { TouchDownArgs, TouchMoveArgs, TouchMoveIosArgs } from './types'
 import type { InitializeTransactionReturn, InstallOptions } from '@/services/core/transaction-service/types'
 
 export class DeviceControlService {
   constructor(
-    private deviceChannel: string,
-    private isDeviceIos: boolean
+    protected deviceBySerialStore: DeviceBySerialStore,
+    private transactionServiceFactory: () => TransactionService
```

Comment:
Any way we can extract the `() => TransactionService` type for the factory?   
I see this is used a lot  
Seems redundant